Dijkstra.java

```diff
@@ -0,0 +1,208 @@
+        visited.add(currentNode);
+
+        // visit the rest of the nodes
+        while(!endFlag) {
+            currentNode = getNext(unvisited);
+            visit(currentNode);
+            unvisited.remove(currentNode);
+            visited.add(currentNode);
+            if (currentNode.getCode().equals(end)){
+                endFlag = true;
+            }
+        }
+
+
+        // output results to the user on the console
+        System.out.println("The shortest path between " + start + " and " + end + " is " + currentNode.getDistance() + " miles.");
+        System.out.println("The shortest path is:");
+        ArrayList path = getPath(visited, start, end);
+        for (int i = path.size() - 1; i > 0; i--){
+            System.out.print(path.get(i) + " --> ");
+        }
+        System.out.print(end);
+
+        // close the keyboard scanner
+        input.close();
+    }
+
+    // visit a Node in the graph. Process the corresponding row in the adjacency matrix
+    // and adjust distances as necessary.
+    // if the distance is adjusted, set the predecessor node to the node being processed
+    public static void visit (Node current){
+        int rowNum = codes.indexOf(current.getCode());
+        ArrayList <Integer> row = distances.get(rowNum);
+        for(Node temp : unvisited){
+            if ((current.getDistance() + row.get(codes.indexOf(temp.getCode()))) < temp.getDistance()) {
+                temp.setDistance(current.getDistance() + row.get(codes.indexOf(temp.getCode())));
+                temp.setPrevNode(current);
+            }
+        }
+    }
+
+    // get the smallest element in the list (next node to visit)
+    public static Node getNext (ArrayList<Node> list){
+        Node temp = list.get(0);
+        for (Node elm : list){
+            if (elm.compareTo(temp) < 1){
+                temp = elm;
+            }
+        }
+        return temp;
+    }
+
+    // generate the path by tracing each node to its predecessor
+    public static ArrayList<String> getPath(ArrayList<Node> n, String st, String en){
+        Node e = null;
+        Node s = null;
+        Node temp = null;
+        ArrayList<String> list = new ArrayList<>();
+        for (Node elm : n){
+            if (elm.getCode().equals(en)){
+                temp = e = elm;
+            }
+            else if (elm.getCode().equals(st)){
+                s = elm;
+            }
+        }
+        while (temp != s){
+            list.add(temp.getCode());
+            temp = temp.getPrevNode();
+        }
+        list.add(s.getCode());
+        return list;
+    }
+
+
+    // a class to represent the vertices in a graph
+    public static class Node implements Comparable<Node>{
+        private String code;
+        private Integer distance;
+        private Node prevNode;
+
+        public Node(){
+            code = "";
+            distance = Integer.MAX_VALUE;
+            prevNode = null;
+        }
+
+        public Node(String c){
```

Comment:
I also extend my previous advice about one-letter-variables to parameters. A lot of times when someone is working with your class they will try and use autofill to figure out what a method takes, and "c" will be of no use to them. So they will have to go find your class and investigate your code to figure out what it is.